### PR TITLE
[N8N-2663] expect multiple answers for survey

### PR DESCRIPTION
**Pull request by saintsebastian**
[N8N-2663](https://linear.app/n8n/issue/N8N-2663/fix-recommendation-logic)

### CODE REVIEW

File: packages/editor-ui/src/components/PersonalizationModal.vue [hunk 1]
```diff
@@ -249,7 +249,7 @@ export default mixins(showMessage, workflowHelpers).extend({
 		closeDialog() {
 			this.modalBus.$emit('close');
 		},
-		onInput(name: IPersonalizationSurveyKeys, value: string) {
+		onInput(name: IPersonalizationSurveyKeys, value: string & string[]) {
```
Comment (mutdmour) [inline packages/editor-ui/src/components/PersonalizationModal.vue:252]: why `&`

File: packages/editor-ui/src/modules/helper.ts [hunk 2]
```diff
@@ -39,36 +48,36 @@ export function getPersonalizedNodeTypes(answers: IPersonalizationSurveyAnswers)
 	}
 
 	if (companySize === COMPANY_SIZE_500_999 || companySize === COMPANY_SIZE_1000_OR_MORE) {
-		if (workArea === SALES_BUSINESSDEV_WORK_AREA) {
+		if (workArea === SALES_BUSINESSDEV_WORK_AREA || findWorkAreaAnswer(SALES_BUSINESSDEV_WORK_AREA)) {
```
Comment (mutdmour) [inline packages/editor-ui/src/modules/helper.ts:51]: maybe we can move the string check into this function instead

File: packages/editor-ui/src/modules/helper.ts [hunk 3]
```diff
@@ -6,7 +6,16 @@ export function getPersonalizedNodeTypes(answers: IPersonalizationSurveyAnswers)
...
 
-	if (companySize === null && workArea === null && answers[CODING_SKILL_KEY] === null) {
+	function findWorkAreaAnswer(name: string) {
+		if (Array.isArray(workArea)) {
+			return workArea.includes(name);
+		}
+		return false;
+	}
+
+	const workAreaIsempty = workArea === null || workArea.length === 0;
```
Comment (mutdmour) [inline packages/editor-ui/src/modules/helper.ts:16]: nitpick - workAreaIsEmpty

File: packages/editor-ui/src/components/PersonalizationModal.vue [hunk 4]
```diff
@@ -90,12 +90,12 @@
...
 					:value="values[OTHER_COMPANY_INDUSTRY_KEY]"
 					placeholder="Specify your company's industry"
-					@input="(value) => onInput(OTHER_COMPANY_INDUSTRY_KEY, value)"
+					@input="(value) => values[OTHER_COMPANY_INDUSTRY_KEY] = value"
 				/>
 
 
 				<n8n-input-label label="How big is your company?">
-					<n8n-select :value="values[COMPANY_SIZE_KEY]" placeholder="Select..." @change="(value) => onInput(COMPANY_SIZE_KEY, value)">
+					<n8n-select :value="values[COMPANY_SIZE_KEY]" placeholder="Select..." @change="(value) => values[COMPANY_SIZE_KEY] = value">
```
Comment (mutdmour) [inline packages/editor-ui/src/components/PersonalizationModal.vue:98]: I know this is easier than adding a helper.. but helpers are much easier to follow... it adds typing which maybe you are trying to escape here.

File: packages/editor-ui/src/components/PersonalizationModal.vue [hunk 5]
```diff
@@ -249,28 +249,19 @@ export default mixins(showMessage, workflowHelpers).extend({
...
 		},
-		onInput(name: IPersonalizationSurveyKeys, value: string) {
-			if (name === WORK_AREA_KEY && value.includes(OTHER_WORK_AREA_OPTION)) {
-				this.otherWorkAreaFieldVisible = true;
+		onMultiInput(name: IPersonalizationSurveyKeys, value: string[]) {
+			if (name === WORK_AREA_KEY) {
+				this.otherWorkAreaFieldVisible = value.includes(OTHER_WORK_AREA_OPTION);
+				this.showAllIndustryQuestions = !value.includes(NOT_APPLICABLE_WORK_AREA);
+				this.values[OTHER_WORK_AREA_KEY] = value.includes(OTHER_WORK_AREA_OPTION) ? this.values[OTHER_WORK_AREA_KEY] : null;
+				this.values[WORK_AREA_KEY] = value;
```
Comment (mutdmour) [inline packages/editor-ui/src/components/PersonalizationModal.vue:257]: nice rewrite

**Review by mutdmour — APPROVED** [review]

**Comment by janober** [comment]
Thanks a lot. Got merged.

**Comment by janober** [comment]
Got released with `n8n@0.153.0`

**Pull request merged**